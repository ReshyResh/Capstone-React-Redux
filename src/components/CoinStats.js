import React from 'react'

const CoinStats = (props) => {
    const { coin } = props;
    return (
      <ul className="stats-list">
        <li>
          <span className="coin-details-title">Price Change:</span>
          <span
            className={
              coin.priceChange > 0
                ? "coin-details-value green"
                : "coin-details-value red"
            }
          >
            {coin.priceChange > 0
              ? "+" + Number(coin.priceChange)
              : Number(coin.priceChange)}{" "}
          </span>
        </li>
        <li>
          <span className="coin-details-title">Price Change %:</span>
          <span
            className={
              coin.priceChangePercent > 0
                ? "coin-details-value green"
                : "coin-details-value red"
            }
          >
            {coin.priceChangePercent > 0
              ? "+" + Number(coin.priceChangePercent) + "%"
              : Number(coin.priceChangePercent) + "%"}
          </span>
        </li>
        <li>
          <span className="coin-details-title">Highest price:</span>
          <span className="coin-details-value">{Number(coin.highPrice)}</span>
        </li>
        <li>
          <span className="coin-details-title">Lowest price:</span>
          <span className="coin-details-value">{Number(coin.lowPrice)}</span>
        </li>
        <li>
          <span className="coin-details-title">Previous close:</span>
          <span className="coin-details-value">
            {Number(coin.prevClosePrice)}
          </span>
        </li>
        <li>
          <span className="coin-details-title">Volume:</span>
          <span className="coin-details-value">{Number(coin.volume)}</span>
        </li>
      </ul>
    );
}

export default CoinStats
