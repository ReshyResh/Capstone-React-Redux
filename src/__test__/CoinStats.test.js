import React from 'react';
import renderer from 'react-test-renderer';
import CoinStats from '../components/CoinStats'
import coin from './MockApiSingle';

it('Coin stats render correctly', () => {
  const tree = renderer
    .create(<CoinStats coin = { coin } />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});