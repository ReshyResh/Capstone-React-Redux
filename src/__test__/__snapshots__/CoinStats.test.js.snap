// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin stats render correctly 1`] = `
<ul
  className="stats-list"
>
  <li>
    <span
      className="coin-details-title"
    >
      Price Change:
    </span>
    <span
      className="coin-details-value green"
    >
      +0.002232
      $
    </span>
  </li>
  <li>
    <span
      className="coin-details-title"
    >
      Price Change %:
    </span>
    <span
      className="coin-details-value green"
    >
      +3.677%
    </span>
  </li>
  <li>
    <span
      className="coin-details-title"
    >
      Highest price:
    </span>
    <span
      className="coin-details-value"
    >
      0.063408
      $
    </span>
  </li>
  <li>
    <span
      className="coin-details-title"
    >
      Lowest price:
    </span>
    <span
      className="coin-details-value"
    >
      0.059927
      $
    </span>
  </li>
  <li>
    <span
      className="coin-details-title"
    >
      Volume:
    </span>
    <span
      className="coin-details-value"
    >
      111634.611
    </span>
  </li>
</ul>
`;
